data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/validator/GetAttestationData.java

```diff
@@ -33,30 +35,53 @@
 import io.javalin.plugin.openapi.annotations.OpenApiResponse;
-import java.util.List;
-import java.util.Map;
 import java.util.Optional;
+import java.util.function.Function;
+import org.jetbrains.annotations.NotNull;
 import tech.pegasys.teku.api.DataProvider;
 import tech.pegasys.teku.api.ValidatorDataProvider;
 import tech.pegasys.teku.api.response.v1.validator.GetAttestationDataResponse;
-import tech.pegasys.teku.api.schema.AttestationData;
-import tech.pegasys.teku.beaconrestapi.handlers.AbstractHandler;
-import tech.pegasys.teku.beaconrestapi.schema.BadRequest;
+import tech.pegasys.teku.beaconrestapi.MigratingEndpointAdapter;
 import tech.pegasys.teku.infrastructure.async.SafeFuture;
+import tech.pegasys.teku.infrastructure.json.types.SerializableTypeDefinition;
+import tech.pegasys.teku.infrastructure.restapi.endpoints.AsyncApiResponse;
+import tech.pegasys.teku.infrastructure.restapi.endpoints.EndpointMetadata;
+import tech.pegasys.teku.infrastructure.restapi.endpoints.ParameterMetadata;
+import tech.pegasys.teku.infrastructure.restapi.endpoints.RestApiRequest;
 import tech.pegasys.teku.infrastructure.unsigned.UInt64;
-import tech.pegasys.teku.provider.JsonProvider;
+import tech.pegasys.teku.spec.datastructures.operations.AttestationData;
 
-public class GetAttestationData extends AbstractHandler {
+public class GetAttestationData extends MigratingEndpointAdapter {
   public static final String ROUTE = "/eth/v1/validator/attestation_data";
-
   private final ValidatorDataProvider provider;
 
-  public GetAttestationData(final DataProvider provider, final JsonProvider jsonProvider) {
-    this(provider.getValidatorDataProvider(), jsonProvider);
+  private static final ParameterMetadata<UInt64> SLOT_PARAM =
+      SLOT_PARAMETER.withDescription(
+          "`uint64` The slot for which an attestation data should be created.");
+
+  private static final SerializableTypeDefinition<AttestationData> RESPONSE_TYPE =
+      SerializableTypeDefinition.object(AttestationData.class)
+          .name("ProduceAttestationDataResponse")
+          .withField(
+              "data", AttestationData.SSZ_SCHEMA.getJsonTypeDefinition(), Function.identity())
+          .build();
+
+  public GetAttestationData(final DataProvider provider) {
+    this(provider.getValidatorDataProvider());
   }
 
-  public GetAttestationData(final ValidatorDataProvider provider, final JsonProvider jsonProvider) {
-    super(jsonProvider);
+  public GetAttestationData(final ValidatorDataProvider provider) {
+    super(
+        EndpointMetadata.get(ROUTE)
+            .operationId("getAttestationData")
+            .summary("Produce an AttestationData")
+            .description("Requests that the beacon node produce an AttestationData.")
+            .tags(TAG_VALIDATOR, TAG_VALIDATOR_REQUIRED)
+            .queryParam(SLOT_PARAM)
+            .queryParam(COMMITTEE_INDEX_PARAMETER)
+            .response(SC_OK, "Request successful", RESPONSE_TYPE)
```

Comment:
service_unavailable missing, i think that's what'll be happening when syncing, so that's why im asking about return code for that other test, it maybe 500 and probably should be 503 according to the api spec.